contracts/eosio.sudo/eosio.sudo.cpp

```diff
@@ -0,0 +1,37 @@
+#include <eosio.sudo/eosio.sudo.hpp>
+#include <eosiolib/transaction.hpp>
+
+namespace eosio {
+
+/*
+exec function manually parses input data (instead of taking parsed arguments from dispatcher)
+because parsing data in the dispatcher uses too much CPU if the included transaction is very big
+
+If we use dispatcher the function signature should be:
+
+void sudo::exec( account_name executer,
+                 transaction  trx )
+*/
+
+void sudo::exec() {
+   require_auth( _self );
+
+   constexpr size_t max_stack_buffer_size = 512;
+   size_t size = action_data_size();
+   char* buffer = (char*)( max_stack_buffer_size < size ? malloc(size) : alloca(size) );
+   read_action_data( buffer, size );
+
+   account_name executer;
+
+   datastream<const char*> ds( buffer, size );
+   ds >> executer;
+
+   require_auth( executer );
+
+   size_t trx_pos = ds.tellp();
+   send_deferred( (uint128_t(executer) << 64) | current_time(), executer, buffer+trx_pos, size );
```

Comment:
Shouldn't it be `size-trx_pos` ?

Reply:
It should be. Thanks.

In practice, `send_deferred` is pretty flexible about the size passed in as long as it isn't smaller than the actual size of the serialized transaction data, but indeed `size-trx_pos` is the correct expression for the argument to pass into this `send_deferred` call.